content/post/against-ab-tests.md

```diff
@@ -0,0 +1,93 @@
+---
+author: Michael Kaminsky
+title: "Against A/B Tests"
+draft: True
+publishDate: "2018-12-01"
+tags:
+ - Statistics
+ - Testing
+ - Web Analytics
+---
+
+The notion of an A/B test is premised on the fundamentally flawed assumption that there is one version of some treatment that is better for all users at all times. Analytics practitioners should embrace this variation and start designing systems that allow for (and encourage) non-binary outcomes of tests.
+
+<!--more-->
+
+In the past few weeks, two really interesting articles about non-standard interpretations of A/B tests have been published. One, from [Uber's engineering blog](https://eng.uber.com/tag/quantile-treatment-effects/), is about calculating "quantile treatment effects" and the other, from [StitchFix's always excellent data science blog](https://multithreaded.stitchfix.com/blog/2018/11/08/bandits/) discusses the use of contextual bandit algorithms to achiever personalization.
+
+Both of these posts are interesting, but I think they both bury the lede a bit as they dive into interpretation and implementation details. I'll restate my thesis for emphasis:
+
+**Traditional A/B testsing rests on a fundamentally flawed premise. Most of the time, version A will be better for some subgroups, and version B will be better for others. Choosing either A or B is _inherently_ inferior to choosing a targeted mix of A _and_ B.**
```

Comment:
This statement, and the premise of this post, are bold! The following specific thoughts come to mind (not looking necessarily for a response, just food for thought).

1. Fundamentally your claim is toying with the notion of objective truths. Basically, if we could create a model with near infinite parameterization, then each human would get their own optimal model. Short of that, we're doing it wrong. I think what you're suggesting is that the current approach of " one size fits all" is suboptimal. So let's add parameters to our models (in this case accounting for *some* heterogeneity in the underlying population) to get one step closer to some utopian world of perfect models.

2. Having said that, practically speaking, we can sometimes stop at the "one size fits all" level. For example, clinical drug testing comes to mind. Sure there's a small portion of the population that has very large side effects when given the polio vaccination, but we probably don't need to optimize for that subgroup when thinking of the vaccine as largely successful.

3. The point above could be answered with "no - in fact we should be optimizing so even the fraction of a % of the population that reacts negatively to the polio vaccine is accounted for with a new medicine". To which I'd say "sure, in theory we should be saving everyone, but in practice we have limited resources. What is the next best use of our human capital?" In this context, I think the onus is on us to show that its a better use of resources to optimize at the sub group level than at the overall level.

Reply:
All these are great responses. I'm intentionally being a bit dramatic for the sake of conversation-starting / click-generating ;)

If this post gets good traction, maybe you / we can follow-up with something that outlines these points and clarifies where this is appropriate.

Also, re: clinical trials, a lot of my thinking on this has been influenced by some of the reading I've done about why clinical trials (as they're mandated by the FDA today) aren't great because they don't take into account plausible response heterogeneity. Scott Alexander at [SSC](http://slatestarcodex.com/) writes about this sometimes and I think it's really fascinating.